diff --git a/config.lzma-linux.h b/config.lzma-linux.h
index e8b00d8..092696f 100644
--- a/config.lzma-linux.h
+++ b/config.lzma-linux.h
@@ -56,7 +56,9 @@
 /* #undef HAVE_COMMONCRYPTO_COMMONDIGEST_H */

 /* Define to 1 if you have the <cpuid.h> header file. */
-#define HAVE_CPUID_H 1
+#ifdef __x86_64__
+   #define HAVE_CPUID_H 1
+#endif

 /* Define if the GNU dcgettext() function is already present or preinstalled.
    */
@@ -309,7 +311,9 @@

 /* Define to 1 if _mm_clmulepi64_si128 is usable. See configure.ac for
    details. */
+#ifdef __x86_64__
 #define HAVE_USABLE_CLMUL 1
+#endif

 /* Define to 1 if you have the `utime' function. */
 /* #undef HAVE_UTIME */
diff --git a/boost/boost.bzl b/boost/boost.bzl
index 8277dbb..afc9569 100644
--- a/boost/boost.bzl
+++ b/boost/boost.bzl
@@ -139,9 +139,9 @@ def boost_deps():
         http_archive,
         name = "org_lzma_lzma",
         build_file = "@com_github_nelhage_rules_boost//:lzma.BUILD",
-        url = "https://github.com/tukaani-project/xz/releases/download/v5.4.4/xz-5.4.4.tar.gz",
-        sha256 = "aae39544e254cfd27e942d35a048d592959bd7a79f9a624afb0498bb5613bdf8",
-        strip_prefix = "xz-5.4.4",
+        url = "https://src.fedoraproject.org/lookaside/extras/xz/xz-5.4.6.tar.gz/sha512/b08a61d8d478d3b4675cb1ddacdbbd98dc6941a55bcdd81a28679e54e9367d3a595fa123ac97874a17da571c1b712e2a3e901c2737099a9d268616a1ba3de497/xz-5.4.6.tar.gz",
+        sha256 = "aeba3e03bf8140ddedf62a0a367158340520f6b384f75ca6045ccc6c0d43fd5c",
+        strip_prefix = "xz-5.4.6",
     )

     maybe(